### PR TITLE
Support old sceNpScoreRankData

### PR DESCRIPTION
**Pull request by RipleyTom**
SceNpScoreRankData was changed starting with SDK 180.002 and some old games use the deprecated structure, this adds support for it.
This also affects SceNpScorePlayerRankData as it encapsulates SceNpScoreRankData.
It doesn't check for SDK version, the library simply uses the submitted array size as a hint to whether it's the new or the old struct.

Fixes https://github.com/RPCS3/rpcs3/issues/15794.

Thanks to @CookiePLMonster for help with this one.

### CODE REVIEW

File: rpcs3/Emu/NP/np_requests.cpp [hunk 1]
```diff
@@ -1086,6 +1088,23 @@ namespace np
 		transaction_async_handler(std::move(lock), trans_ctx, req_id, async);
 	}
 
+	template <typename T>
+	void set_rankdata_values(T& cur_rank, const ScoreRankData* fb_rankdata)
```
Comment (CookiePLMonster) [inline rpcs3/Emu/NP/np_requests.cpp:1092]: Consider turning it into `vm::ptr<T>& cur_rank` and then the `is_same_v` check below can be simpler.

File: rpcs3/Emu/NP/np_contexts.h [hunk 2]
```diff
@@ -77,6 +77,8 @@ struct tdata_get_score_generic
 	u32 arrayNum = 0;
 	vm::ptr<CellRtcTick> lastSortDate;
 	vm::ptr<SceNpScoreRankNumber> totalRecord;
+	bool player_rank_data;
```
Comment (Megamouse) [inline rpcs3/Emu/NP/np_contexts.h:80]: These should both be default inited

**Comment by CookiePLMonster** [comment]
As per https://github.com/RipleyTom/rpcn/pull/77, once this PR is merged, the board ID 0 for Ridge Racer 7 (`NPWR00001`) needs to be wiped, as currently it contains only incorrect zero entries.

**Review by CookiePLMonster — APPROVED** [review]
Works well, both Gran Turismo HD Concept and Ridge Racer 7 can now retrieve leaderboards correctly. Also fixes RR7's time trial submissions, as previously the game refused to submit them due to the retrieval failure.

GTHD's leaderboards still don't function fully correctly, but that is unrelated to the PR.

**Pull request merged**